### PR TITLE
Handle cases where no rate limiting is used

### PR DESCRIPTION
**Pull request by augustd**
In this case I catch the exception that is thrown and create a rate limit object with significantly high (1 billion) remaining requests. The rest of the code operates normally.

### CODE REVIEW

**Comment by augustd** [comment]
Fixes #61

**Comment by augustd** [comment]
A nice addition to this would be to suppress/ignore/auto answer 'yes' to the `Do you want to continue? [y/N]:` message.

File: ghsearch/gh_search.py [hunk 1]
```diff
@@ -52,8 +53,19 @@ def __init__(self, client: github.Github, filters: List[Filter], verbose: bool =
         self.filters = filters
         self.verbose = verbose
 
+    def get_rate_limit(self):
+        try:
+            return self.client.get_rate_limit()
+        except Exception as e:
+            # Handle the exception by returning a default rate limit dictionary
```
Comment (janeklb) [inline ghsearch/gh_search.py]: Can the exception type be scoped down to something more specific?

Comment (janeklb) [reply ghsearch/gh_search.py]: (i've updated the PyGithub dependency -- i suggest rebasing your PR branch from main to see what the latest version does)

**Review by janeklb — CHANGES_REQUESTED** [review]
Hi @augustd - thank you for the contributions (just noticed them now!)

**Review by janeklb — APPROVED** [review]

**Comment by janeklb** [comment]
@augustd are you able to verify that the changes I've made work for you?

**Pull request merged**